src/goto-instrument/dump_c_class.h

```diff
@@ -21,6 +21,120 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include <goto-programs/system_library_symbols.h>
 
+/// Used for configuring the behaviour of dump_c
+struct dump_c_configurationt final
+{
+  /// Include the function declarations in the dump
+  bool include_function_decls;
+
+  /// Include the functions in the dump
+  bool include_function_bodies;
+
+  /// Include the global declarations in the dump
+  bool include_global_decls;
+
+  /// Include the typedefs in the dump
+  bool include_typedefs;
+
+  /// Include global variable definitions in the dump
+  bool include_global_vars;
+
+  /// Include struct definitions in the dump
+  bool include_compounds;
+
+  /// Define whether to follow compunds recursively
+  bool follow_compounds;
+
+  /// Include headers type declarations are borrowed from
+  bool include_headers;
+
+  dump_c_configurationt()
+    : include_function_decls(true),
+      include_function_bodies(true),
+      include_global_decls(true),
+      include_typedefs(true),
+      include_global_vars(true),
+      include_compounds(true),
+      follow_compounds(true),
+      include_headers(false)
+  {
+  }
+
+  dump_c_configurationt(
+    const bool include_function_decls,
+    const bool include_function_bodies,
+    const bool include_global_decls,
+    const bool include_typedefs,
+    const bool include_global_vars,
+    const bool include_compounds,
+    const bool follow_compounds,
+    const bool include_headers)
+    : include_function_decls(include_function_decls),
+      include_function_bodies(include_function_bodies),
+      include_global_decls(include_global_decls),
+      include_typedefs(include_typedefs),
+      include_global_vars(include_global_vars),
+      include_compounds(include_compounds),
+      follow_compounds(follow_compounds),
+      include_headers(include_headers)
```

Comment:
I'm not sure this constructor is useful since it seems clearer to initialize a default one and then set the options as needed.